manual/tuples/README.md

```diff
@@ -1,5 +1,77 @@
-## Tuples
+## Using Tuples with the Java driver
 
-*Coming soon... In the meantime, see the javadoc for [TupleType].*
+Cassandra allows to use `tuple` data types [in tables](https://docs.datastax.com/en/cql/3.1/cql/cql_reference/tupleType.html):
+
+```
+CREATE TABLE ks.collect_things (
+  pk int,
+  ck1 text,
+  ck2 text,
+  v tuple<int, text, float>,
+  PRIMARY KEY (pk, ck1, ck2)
+);
+```
+
+### Fetching Tuples from Rows results
+
+The DataStax Java driver exposes a special `TupleValue` class to handle such columns. `TupleValue` extends `GettableByIndexData` class, which allows to call specific `get...(int)` methods on a returned `TupleValue`:
+
+```java
+Row row = session.execute("SELECT v FROM ks.collect_things WHERE pk = 1").one();
+
+TupleValue tupleValue = row.getTupleValue("v");
+
+int firstValueInTuple = tupleValue.getInt(0);
+
+String secondValueInTuple = tupleValue.getString(1);
+
+Float thirdValueInTuple = tupleValue.getFloat(2);
+```
+
+### Using tuples as statement parameters
+
+A prepared statement may be containing a Tuple as a query parameter. In such case, users will need to create or gather a `TupleType` first, in order to be able to create a `TupleValue` to bind:
+
+```java
+PreparedStatement ps = session.prepare("INSERT INTO ks.collect_things (pk, ck1, ck2, v) VALUES (:pk, :ck1, :ck2, :v)");
+
+TupleType tupleType = cluster.getMetadata().newTupleType(DataType.cint(), DataType.text(), DataType.cfloat());
+
+BoundStatement bs = ps.bind();
+bs.setInt("pk", 1);
+bs.setString("ck1", "1");
+bs.setString("ck2", "1");
+bs.setTupleValue("v", tupleType.newValue(1, "hello", 2.3f));
+
+session.execute(bs);
+```
+
+#### More use cases
+
+Users can also define single-usage tuples in _SELECT_ queries with the `IN` keyword, usually for tables with composite clustering keys, in that case a tuple will be usable the same way it was for prepared statements parameters:
```

Comment:
This is called a "multi-column IN restriction" – maybe it's worth mentioning this term somewhere.